frontend/src/component/admin/users/UsersList/UsersActionsCell/UsersActionsCell.tsx

```diff
@@ -1,101 +1,173 @@
+                            Access overview
+                        </UserAction>
+                    )}
+                    <UserAction
+                        onClick={() => {
+                            onChangePassword();
+                            handleClose();
+                        }}
+                        permission={ADMIN}
+                        isScimUser={isScimUser}
+                    >
+                        Change password
+                    </UserAction>
+                    <UserAction
+                        onClick={() => {
+                            onResetPassword();
+                            handleClose();
+                        }}
+                        permission={ADMIN}
+                        isScimUser={isScimUser}
+                    >
+                        Reset password
+                    </UserAction>
+                    <UserAction
+                        onClick={() => {
+                            onDelete();
+                            handleClose();
                         }}
+                        permission={ADMIN}
                     >
-                        <Key />
-                    </PermissionIconButton>
-                }
-            />
+                        Remove user
+                    </UserAction>
+                </MenuList>
+            </StyledPopover>
+        </StyledActions>
+    );
+};
 
-            <PermissionIconButton
-                data-loading
-                onClick={onChangePassword}
-                permission={ADMIN}
-                tooltipProps={{
-                    title: isScimUser ? scimTooltip : 'Change password',
-                }}
-                disabled={isScimUser}
-            >
-                <Lock />
-            </PermissionIconButton>
-            <PermissionIconButton
-                data-loading
-                onClick={onResetPassword}
-                permission={ADMIN}
-                tooltipProps={{
-                    title: isScimUser ? scimTooltip : 'Reset password',
-                }}
-                disabled={isScimUser}
-            >
-                <LockReset />
-            </PermissionIconButton>
-            <PermissionIconButton
-                data-loading
-                onClick={onDelete}
-                permission={ADMIN}
-                tooltipProps={{
-                    title: 'Remove user',
-                }}
-            >
-                <Delete />
-            </PermissionIconButton>
-        </StyledBox>
+interface IUserActionProps {
+    onClick: () => void;
+    permission: string;
+    isScimUser?: boolean;
+    children: React.ReactNode;
+}
+
+const UserAction = ({
+    onClick,
+    permission,
+    isScimUser,
+    children,
+}: IUserActionProps) => {
+    const scimTooltip =
+        'This user is managed by your SCIM provider and cannot be changed manually';
+
+    return (
+        <PermissionHOC permission={permission}>
```

Comment:
I noticed the users page is only accessible to ADMINs. That means we can simplify this further.

Reply:
Addressed in https://github.com/Unleash/unleash/pull/9525/commits/2cee099b70b42afdd2f9a02ae58b99d075783d23